### ISSUE TITLE
My model keeps predicting only one label out of many labels (Not overfitting).

### ISSUE THREAD

**Issue by Daocuong-main** [tensorflow/federated#3327]
**Describe the bug**
I'm trying to build a model for a "Traffic classification" problem for a computer network. I initially used LSTM + CNN together with our dataset and the continuous model gave very low accuracy on the test set. I thought my model was overfitting, so I tried many ways to deal with the overfitting problem but did not bring the expected results.
So I tried printing out what the model predicted and found out that the model only predicted a single label out of many. So I tried with another model, another dataset, namely the Linear model with the iris dataset. The results are the same no matter how good the training model is.
From there I started to suspect tff dividing my data to the clients but I don't know where the problem is. I noticed that with parameter ```PREFETCH_BUFFER = 1``` the model predicts 1 label, when I adjust ```PREFETCH_BUFFER = 3``` the model still predicts 1 label, but when I adjust ```PREFETCH_BUFFER = 5``` the model predicts 2 labels and predicts quite randomly, but the model should predict 3 labels.

I put the link to my notebook here [jupyter notebook](https://github.com/Daocuong-main/FL/blob/main/Linear_FL_Iris.ipynb) and my model [Linear.py](https://github.com/Daocuong-main/FL/blob/main/models/Linear.py)

**Environment:**
* OS Platform and Distribution: Linux Ubuntu 18.04
* Python package versions (e.g., TensorFlow Federated, TensorFlow):
  - tensorflow==2.10.0
  - tensorflow-federated==0.36.0
* Python version: 3.9.14
* CUDA/cuDNN version: No CUDA, cpu only

**Expected behavior**
I expect my model to predict well on the test set and to predict all the labels.

**Additional context**
When I try to print it out and compare my "sample_batch" with the example code in the [Federated Learning for Image Classification tutorial](https://www.tensorflow.org/federated/tutorials/federated_learning_for_image_classification#preprocessing_the_input_data):
* My code:
```
def preprocess(dataset):

    def batch_format_fn(element):
        """Flatten a batch `pixels` and return the features as an `OrderedDict`."""

        return collections.OrderedDict(
            x=reshape(element['pixels'], [-1, 1, NUM_FEATURE]),
            y=reshape(element['label'], [-1, 1]))

    return dataset.repeat(NUM_EPOCHS).shuffle(SHUFFLE_BUFFER).batch(
        BATCH_SIZE).map(batch_format_fn).prefetch(PREFETCH_BUFFER)


preprocessed_sample_dataset = preprocess(sample_dataset)
sample_batch = nest.map_structure(
    lambda x: x.numpy(), next(iter(preprocessed_sample_dataset)))
```
Output:
```OrderedDict([('x', array([[[6.3, 2.3, 4.4, 1.3]]])), ('y', array([[1]]))])```
* Example code:
```
preprocessed_example_dataset = preprocess(example_dataset)

sample_batch = tf.nest.map_structure(lambda x: x.numpy(),
                                     next(iter(preprocessed_example_dataset)))

sample_batch
```
Output:
```
OrderedDict([('x', array([[1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       ...,
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.]], dtype=float32)), ('y', array([[2],
       [1],
       [5],
       [7],
       [1],
       [7],
       [7],
       [1],
       [4],
       [7],
       [4],
       [2],
       [2],
       [5],
       [4],
       [1],
       [1],
       [0],
       [0],
       [9]], dtype=int32))])
```
It can be seen that I do the same with the example code but the result is quite different, with my code there is only one data sample, but with the example code there are many data samples and many labels. I don't know why either.

**Comment by michaelreneer** [comment]
This doesn't seem like a TFF bug? Maybe asking on StackOverflow would widen the audience?
